server/publicapi/items/service.py

```diff
@@ -93,7 +94,10 @@ def get(self, req, lookup):
         req.args['filter'] = json.dumps(date_filter)
         self._set_fields_filter(req)  # Eve's "projection"
 
-        return super().get(req, lookup)
+        try:
+            return super().get(req, lookup)
+        except InvalidSearchString:
+            raise BadParameterValueError('invalid search text')
```

Comment:
Details on _why_ the search text was invalid would be helpful to the user, please check if we can use something meaningful from the `InvalidSearchString`'s error message.